<launch>

	<!-- default arguments -->
	<arg name="uart_comm_running" default="false" />
    <arg name="pres_ctrl_params" default="pres_ctrl_params_default.yaml" />

	<!-- load control parameters -->
    <rosparam command="load" file="$(find mcu_uart)/config/$(arg pres_ctrl_params)" />

	<!-- run UART communication if not already running -->
	<node
		unless="$(arg uart_comm_running)"
        name="uart_comm"
        pkg="mcu_uart" 
        type="uart_comm" 
        output="screen"
    /> 

	<!-- run MCU value configuration -->
	<node
        name="config_vals"
        pkg="mcu_uart" 
        type="config_vals.py" 
        output="screen"
    /> 
    
</launch>
